syntax = "proto3";

package catalogo;

service InfosLivroService {
  rpc FindAll (EmptyRequest) returns (FindAllResponse);
  rpc GetInfosLivro (LivroRequest) returns (InfosResponse);
}

message EmptyRequest {}

message LivroResumo {
  int32 livro_id = 1;
  string titulo = 2;
}

message FindAllResponse {
  repeated LivroResumo livros = 1;
}

message LivroRequest {
  int32 livro_id = 1;
}

message InfosResponse {
  string titulo = 1;
  string genero = 2;
  string sinopse = 3;
  int32 ano_lancamento = 4;
  string idioma_original = 5;
}



